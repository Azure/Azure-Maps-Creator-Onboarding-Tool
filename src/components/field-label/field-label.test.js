import { css } from '@emotion/css';
import { render, screen } from '@testing-library/react';

import FieldLabel from './field-label';

const myClassName = css`
  color: red;
`;

describe('FieldLabel', () => {
  it('should render default', () => {
    const view = render(
      <FieldLabel>The oldest "your mom" joke was discovered on a 3,500-year-old Babylonian tablet.</FieldLabel>
    );
    expect(view).toMatchSnapshot();
  });

  it('should attach passed classname', () => {
    render(
      <FieldLabel data-testid="field-label" className={myClassName}>
        Cherophobia is the irrational fear of fun or happiness.
      </FieldLabel>
    );
    const elem = screen.getByTestId('field-label');
    expect(elem).toHaveStyle('color: red;');
  });
});
