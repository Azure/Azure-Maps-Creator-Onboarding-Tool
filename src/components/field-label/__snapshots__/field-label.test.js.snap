// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldLabel should render default 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  height: 1.5rem;
  line-height: 1.5rem;
  max-width: 15rem;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.813rem;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        The oldest "your mom" joke was discovered on a 3,500-year-old Babylonian tablet.
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  height: 1.5rem;
  line-height: 1.5rem;
  max-width: 15rem;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.813rem;
}

<div>
    <div
      class="emotion-0"
    >
      The oldest "your mom" joke was discovered on a 3,500-year-old Babylonian tablet.
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
